modules/contact/test/tests/fieldsplit/frictional_mortar_FS.i

```diff
@@ -0,0 +1,183 @@
+offset = 0.021
+
+vy = 0.15
+vx = 0.04
+
+refine = 1
+
+[GlobalParams]
+  displacements = 'disp_x disp_y'
+  volumetric_locking_correction = true
+[]
+
+[Mesh]
+  [./original_file_mesh]
+    type = FileMeshGenerator
+    file = long_short_blocks.e
+  [../]
+  uniform_refine =  ${refine}
+[]
+
+[Modules/TensorMechanics/Master]
+  [./all]
+    strain = FINITE
+    incremental = true
+    add_variables = true
+    block = '1 2'
+  [../]
+[]
+
+[Functions]
+  [./horizontal_movement]
+    type = ParsedFunction
+    value = 'if(t<0.5,${vx}*t-${offset},${vx}-${offset})'
+  [../]
+  [./vertical_movement]
+    type = ParsedFunction
+    value = 'if(t<0.5,${offset},${vy}*(t-0.5)+${offset})'
+  [../]
+[]
+
+[BCs]
+  [./push_left_x]
+    type = FunctionDirichletBC
+    variable = disp_x
+    boundary = 30
+    function = horizontal_movement
+  [../]
+  [./fix_right_x]
+    type = DirichletBC
+    variable = disp_x
+    boundary = 40
+    value = 0.0
+  [../]
+  [./fix_right_y]
+    type = DirichletBC
+    variable = disp_y
+    boundary = '40'
+    value = 0.0
+  [../]
+  [./push_left_y]
+    type = FunctionDirichletBC
+    variable = disp_y
+    boundary = '30'
+    function = vertical_movement
+  [../]
+[]
+
+[Materials]
+  [./elasticity_tensor_left]
+    type = ComputeIsotropicElasticityTensor
+    block = 1
+    youngs_modulus = 1.0e6
+    poissons_ratio = 0.3
+  [../]
+  [./stress_left]
+    type = ComputeFiniteStrainElasticStress
+    block = 1
+  [../]
+
+  [./elasticity_tensor_right]
+    type = ComputeIsotropicElasticityTensor
+    block = 2
+    youngs_modulus = 1.0e6
+    poissons_ratio = 0.3
+  [../]
+  [./stress_right]
+    type = ComputeFiniteStrainElasticStress
+    block = 2
+  [../]
+[]
+
+[Contact]
+  [leftright]
+    mesh = original_file_mesh
+    slave = 10
+    master = 20
+
+    model = coulomb
+    formulation = mortar
+    system = constraint
+
+    friction_coefficient = 0.2
+  [../]
+[]
+
+[ICs]
+  [./disp_y]
+    block = 1
+    variable = disp_y
+    value = ${offset}
+    type = ConstantIC
+  [../]
+  [./disp_x]
+    block = 1
+    variable = disp_x
+    value = -${offset}
+    type = ConstantIC
+  [../]
+[]
+
+[Preconditioning]
+  [./FSP]
+    type = FSP
+    # It is the starting point of splitting
+    topsplit = 'contact_interior' # 'contact_interior' should match the following block name
+    [./contact_interior]
+      splitting          = 'contact interior'
+      splitting_type     = multiplicative
+    [../]
+    [./interior]
+      type = ContactSplit
+      vars = 'disp_x disp_y'
+      uncontact_master   = '20'
+      uncontact_slave    = '10'
+      uncontact_displaced = '30'
+      blocks              = '1 2'
+      include_all_contact_nodes = 1
+
+      petsc_options_iname = '-ksp_type -pc_type -pc_hypre_type '
+      petsc_options_value = '  preonly hypre  boomeramg'
```

Comment:
HYPRE is more efficient when you use it as a preconditioner.

Reply:
Huh? Isn't that what she's doing?

Reply:
I meant: inside of FS, you still can use KSP combined with PC that make the entire solver more robust. In some cases, we may even need to do iteration between blocks.

Reply:
> I meant: inside of FS, you still can use KSP combined with PC that make the entire solver more robust.

Interesting.. I am going to try using KSP inside FS.

>  In some cases, we may even need to do iteration between blocks.

 Isn't FS inherently iterating between blocks?

Reply:
One iteration only right now :)

Reply:
> Interesting.. I am going to try using KSP inside FS

Remember to set the number of iterations as 2 or 3 ... when try this. It is problem-dependent. But do not worry about this at this point

Reply:
@fdkong Now I see what you mean here... Yea I agree including more iterations would definitely improve the robustness of the preconditioner. Thanks for pointing this out!